splitio/sync/split.py

```diff
@@ -351,24 +353,21 @@ def _synchronize_json(self):
         :return: segment names string array
         :rtype: [str]
         """
-        fetched, since, till = self._read_splits_from_json_file(self._filename)
+        fetched, till = self._read_splits_from_json_file(self._filename)
         segment_list = set()
-        if self._split_storage.get_change_number() <= till:
-            to_delete = []
-            if since == -1:
-                to_delete = [name for name in self._split_storage.get_split_names()
-                            if name not in json.dumps(fetched)]
-            for split in fetched:
-                parsed = splits.from_raw(split)
-                _LOGGER.debug("split %s is updated", parsed.name)
-                self._split_storage.put(parsed)
-
-                segment_list.update(set(parsed.get_segment_names()))
-
-            for split in to_delete:
-                self._split_storage.remove(split)
+        if self._get_sha(json.dumps(fetched)) != self._current_json_sha:
```

Comment:
getting `self._get_sha(json.dumps(fetched)` on a variable will be better than calling twice